client/state/media/thunks/create-transient-media-items.js

```diff
@@ -39,7 +39,7 @@ export function createTransientMediaItems( files, site ) {
 
 			dispatch( createMediaItem( site, transientMedia ) );
 
-			return transientMedia;
+			return [ file, transientMedia ];
```

Comment:
How is a tuple better or more readable than an object with `file` and `transientMedia` properties? I think this is an artifact of the prior `_.zip()` usage, but could be more readable if we used an object instead.